File: app/src/main/java/com/shop440/Utils/Application.java
package com.shop440.Utils;

import android.content.Context;
import android.graphics.Typeface;
import android.support.multidex.MultiDex;
import android.util.Log;

import com.facebook.FacebookSdk;
import com.facebook.accountkit.AccountKit;
import com.facebook.appevents.AppEventsLogger;
import com.shop440.R;

/**
 * Created by smilecs on 6/5/15.
 */
public class Application extends android.app.Application {
    private static Application sInstance;
    private static AppEventsLogger logger;
    private static String PATH = "fonts/";

    public static Application getsInstance() {
        return sInstance;
    }

    public static Context getAppContext() {
        return sInstance.getApplicationContext();
    }

    public static AppEventsLogger logger() {
        return logger;
    }

    public static Typeface getFont(int id) {
        Typeface typeface = null;
        String fontId = sInstance.getResources().getStringArray(R.array.fonts)[id];
        fontId = PATH.concat(fontId);
        try {
            typeface = Typeface.createFromAsset(sInstance.getAssets(), fontId);
        } catch (Exception e) {
            Log.d("App", "Could not create font: " + fontId);
        }
        return typeface;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        sInstance = this;
        FacebookSdk.sdkInitialize(getApplicationContext());
        AccountKit.initialize(getApplicationContext());
        AppEventsLogger.activateApp(this);
        logger = AppEventsLogger.newLogger(this);
    }

}

File: app/src/main/java/com/shop440/Application.kt
package com.shop440

import android.content.Context
import android.graphics.Typeface
import android.support.multidex.MultiDex
import android.util.Log
import com.bentech.android.appcommons.AppCommons
import com.facebook.FacebookSdk
import com.facebook.accountkit.AccountKit
import com.facebook.appevents.AppEventsLogger
import com.shop440.api.Urls
import io.realm.Realm

/**
 * Created by smilecs on 6/5/15.
 */
class Application : android.app.Application() {

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
        AppCommons.install(this)
    }

    override fun onCreate() {
        super.onCreate()
        sInstance = this
        Realm.init(this)
        FacebookSdk.sdkInitialize(applicationContext)
        AccountKit.initialize(applicationContext)
        AppEventsLogger.activateApp(this)
        logger = AppEventsLogger.newLogger(this)
        AppCommons.getAppCommonsConfiguration().editTextInvalidEmailErrorMessage = R.string.email_invalid_error
        AppCommons.getAppCommonsConfiguration().editTextPhoneNumberValidatorErrorMessage = R.string.phone_invalid_error_message
        AppCommons.getAppCommonsConfiguration().editTextRequiredErrorMessage = R.string.required_field_error
    }

    companion object {
        private var sInstance: Application? = null
        private var logger: AppEventsLogger? = null
        private val PATH = "fonts/"

        fun getsInstance(): Application? = sInstance

        val appContext: Context
            get() = sInstance?.applicationContext!!

        fun logger(): AppEventsLogger? = logger

        fun getFont(id: Int): Typeface? {
            var typeface: Typeface? = null
            var fontId = sInstance?.resources!!.getStringArray(R.array.fonts)[id]
            fontId = PATH + fontId
            try {
                typeface = Typeface.createFromAsset(sInstance!!.assets, fontId)
            } catch (e: Exception) {
                Log.d("App", "Could not create font: " + fontId)
            }
            return typeface
        }

        val authToken: String
            get() {
                val preferences = getsInstance()!!.getSharedPreferences(getsInstance()!!.getString(R.string.shop440), Context.MODE_PRIVATE)
                return preferences.getString(Urls.TOKEN, "null")
            }
    }


}
